### PR TITLE
DAOS-13593 mercury: update to 2.3.0 final

### PR DESCRIPTION
**Pull request by soumagne**
Add hg_info tool

Fix pie flag passed to binaries on CentOS7

Remove na_ucx_src_port.patch and old patches

Fix license

### CODE REVIEW

File: debian/libmercury1.dirs [hunk 1]
```diff
@@ -1 +1,2 @@
+usr/bin
```
Comment (brianjmurrell) [inline debian/libmercury1.dirs:1]: This should go into a `mercury.dirs` file.  The `libmercury1` package is for the shared library only.

File: debian/libmercury1.install [hunk 2]
```diff
@@ -1 +1,2 @@
+usr/bin/*
```
Comment (brianjmurrell) [inline debian/libmercury1.install:1]: This should go into a `mercury.dirs` file.  The `libmercury1` package is for the shared library only.

File: mercury.spec [hunk 3]
```diff
@@ -91,13 +81,18 @@ Mercury plugin to support the UCX transport.
 %autosetup -p1 -n mercury-%dl_version
 
 %build
+# Pass -pie linker flags manually on CentOS7 due to CMake PIE issues
+# see: https://cmake.org/cmake/help/latest/policy/CMP0083.html#policy:CMP0083
 %cmake  -DCMAKE_IN_SOURCE_BUILD:BOOL=ON                   \
         -DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo          \
+%if 0%{?rhel} < 8
```
Comment (brianjmurrell) [inline mercury.spec:88]: This will match non EL platforms also, such as SUSE, etc.

```suggestion
%if 0%{?rhel} && 0%{?rhel} < 8
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
We really should be running `lintian` on the produced `.deb`s, the same way we run `rpmlint`. That would have caught the binaries in the `libmercury1` package issue.

File: mercury.spec (hunk 3 above)
Comment (soumagne) [reply mercury.spec:88]: ah good catch, thanks!

File: debian/libmercury1.dirs (hunk 1 above)
Comment (soumagne) [reply debian/libmercury1.dirs:1]: ok thanks! We have a similar issue with the libfabric package btw, I was copying what was done there.

Comment (brianjmurrell) [reply debian/libmercury1.dirs]: We should try to fix that on the next round of updates there.  I am doing one for EL9 so I'll take a look at it then.  Or if you do an update there before I can land mine, please do fix that.

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**